src/Facade/Services/RequisitionFacadeService.cs

```diff
@@ -96,6 +96,29 @@ public async Task<IResult<RequisitionHeaderResource>> CancelLine(
             }
         }
 
+        public async Task<IResult<RequisitionHeaderResource>> BookRequisition(int reqNumber, int? lineNumber, int bookedBy, IEnumerable<string> privileges)
+        {
+            try
+            {
+                var privilegeList = privileges.ToList();
+
+                var req = await this.requisitionService.BookRequisition(
+                    reqNumber,
+                    lineNumber,
+                    new User
+                    {
+                        UserNumber = bookedBy,
+                        Privileges = privilegeList
+                    });
+                return new SuccessResult<RequisitionHeaderResource>(
+                    this.BuildResource(req, privilegeList));
+            }
+            catch (Exception e)
```

Comment:
should only catch DomainException so we don't accidentally bury bad news

Reply:
or you could catch DomainException first and then have this general catch fallback afterwards.. only if you do need to handle other exception types in a special way that is. I think the UseExceptionHandler middleware just returns a 500 for any unhandled exceptions anyway and sticks the details in the response body.

Could also wrap all your domain code in a general try catch, and then if anything deeper throws a funky exception type  then catch it and transform it to into a DomainException

Reply:
maybe thats no better than the general Exception catch you had before... not sure